app/controllers/ChargeSummaryController.scala

```diff
@@ -45,78 +47,88 @@ class ChargeSummaryController @Inject()(authenticate: AuthenticationPredicate,
-									case chargeHistory: ChargesHistoryModel =>
-										val documentDetailWithDueDate: DocumentDetailWithDueDate = success.findDocumentDetailByIdWithDueDate(id).get
-										if (isEnabled(TxmEventsApproved)) {
-											auditingService.extendedAudit(ChargeSummaryAudit(
-												mtdItUser = user,
-												docDateDetail = documentDetailWithDueDate,
-												None
-											))
-										}
-										Ok(view(
-											documentDetail = documentDetail,
-											dueDate = success.getDueDateFor(documentDetail),
-											backLocation = backLocation,
-											taxYear = taxYear,
-											chargesHistory = chargeHistory.chargeHistoryDetails.getOrElse(List()),
-											chargeHistoryEnabled = true,
-											latePaymentInterestCharge = isLatePaymentCharge
-										))
-									case _ =>
-										Logger.warn("[ChargeSummaryController][showChargeSummary] Invalid response from charge history")
-										itvcErrorHandler.showInternalServerError()
-								}
-							} else {
-								val documentDetailWithDueDate: DocumentDetailWithDueDate = success.findDocumentDetailByIdWithDueDate(id).get
-								if (isEnabled(TxmEventsApproved)) {
-									auditingService.extendedAudit(ChargeSummaryAudit(
-										mtdItUser = user,
-										docDateDetail = documentDetailWithDueDate,
-										None
-									))
-								}
-								Future.successful(Ok(view(
-									documentDetail = documentDetail,
-									dueDate = success.getDueDateFor(documentDetail),
-									backLocation = backLocation,
-									taxYear = taxYear,
-									chargesHistory = List(),
-									chargeHistoryEnabled = false,
-									latePaymentInterestCharge = isLatePaymentCharge
-								)))
-							}
-            //Should not happen unless url is changed manually so redirect to home
-            case _: FinancialDetailsModel =>
-              Logger.warn(s"[ChargeSummaryController][showChargeSummary] Transaction id not found for tax year $taxYear")
-              Future.successful(Redirect(controllers.routes.HomeController.home().url))
-            case _ =>
-              Logger.warn("[ChargeSummaryController][showChargeSummary] Invalid response from financial transactions")
-              Future.successful(itvcErrorHandler.showInternalServerError())
-          }
+        financialDetailsService.getFinancialDetails(taxYear, user.nino).flatMap {
+          case success: FinancialDetailsModel if success.documentDetails.exists(_.transactionId == id) =>
+            val backLocation = user.session.get(SessionKeys.chargeSummaryBackPage)
+            val documentDetail = success.documentDetails.find(_.transactionId == id).get
+
+            val paymentAllocationEnabled: Boolean = isEnabled(PaymentAllocation)
+            val paymentAllocations: List[PaymentsWithChargeType] =
+              if (paymentAllocationEnabled) {
+                success.financialDetails.filter(_.transactionId.contains(id)).flatMap(_.allocation)
```

Comment:
Can you change this to '==' rather than 'contains' as there could be 'id' which partially match other id's which we don't know?

Reply:
My bad, as I thought transactionId is a string.